### ISSUE TITLE
Run JavaScript snippet after fully booting up Discord

### ISSUE THREAD

**Issue by CodyNinja1** [Vencord/plugin-requests#216]
### Plugin Description

This plugin would allow you to add multiple JS snippets, and would run them when Discord has booted up fully (kinda like pasting them to the console and hitting enter)

I asked already if a plugin like this is allowed, I (and other people) think that this plugin does NOT break TOS.

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by lewisakura** [comment]
what snippets inspired you to want this plugin? it seems more appropriate to make them actual plugins if they're useful rather than have a generic javascript runner that loads arbitrary code. plus, snippets tend to break a lot.

**Comment by CodyNinja1** [comment]
These ones I personally use:
- (Add images to favourites)[https://discord.com/channels/603970300668805120/1117414445036081152] (Discord Previews server)
- (Lurk a guild)[https://discord.com/channels/1015060230222131221/1028106792737185842/1100538860750000168] (Vendor server)

**Comment by Sqaaakoi** [comment]
These should be submitted as plugins

Favourites:
- Request #13
- Plugin PR: https://github.com/Vendicated/Vencord/pull/2091

Guild lurking:
This should be a plugin. I have had some ideas before, see https://canary.discord.com/channels/1015060230222131221/1026515880080842772/1216328731077902346
> A persistent lurk after you click off the guild and on invite links plugin would be a good idea though imo

**Comment by CodyNinja1** [comment]
I personally think there should be a single snippet plugin, since these features are so small that only a few people would use them. I see no point in adding these snippets as plugins, as it would increase the size of Vencord and will get repetitive when new snippets are added.
And what if a new snippet shows up? 
Would you rather wait for someone to PR it, wait for it to get accepted (which doesn't happen all the time) and then update Vencord, or just add the snippet to the plugin and reload discord (at most)?

**Comment by Sqaaakoi** [comment]
> I personally think there should be a single snippet plugin, since these features are so small that only a few people would use them.
vencord already has extremely niche plugins
> I see no point in adding these snippets as plugins, as it would increase the size of Vencord and will get repetitive when new snippets are added.
userplugins?
> And what if a new snippet shows up? Would you rather wait for someone to PR it, wait for it to get accepted (which doesn't happen all the time)
Ideally people would PR their plugins, but speaking from personal experience, I don't PR everything and leave it in a state of "i forgot I even did this" or "what if it isn't up to quality standards"
> and then update Vencord, or just add the snippet to the plugin and reload discord (at most)?
Vencord should NOT try to be BetterDiscord, and that's a good thing IMO. I do remember @Inbestigator had a concept for importing external plugins, which (from my memory) was closed for being not very compatible with Vencord's design. I feel such an idea is a GOOD thing, but ONLY in the case of users using Dev builds already. Such a UI shouldn't show up for users not using a dev build, and that UI should probably be implemented in some way that lets you import from userplugins posted directly in Discord, rather than pasting a link to an external source.

@Inbestigator I'd like your thoughts on this

**Comment by Inbestigator** [comment]
Yeah, I was kinda struggling with how the imports would work, my idea was that it'd be a github repo with just an `index.js` and some sort of identifier script, that way the client could import extra packages if needed. Doing it this way would be problematic bc you would Node to be able to use the plugin.

I still think that being able to "build" a plugin and distribute it would be amazing. Like where the built plugin just gets slapped into the Webpack somehow.

**Comment by Sqaaakoi** [comment]
> Yeah, I was kinda struggling with how the imports would work, my idea was that it'd be a github repo with just an `index.js` and some sort of identifier script, that way the client could import extra packages if needed. Doing it this way would be problematic bc you would need Node to be able to use the plugin.
> 
> I still think that being able to "build" a plugin and distribute it would be amazing. Like where the built plugin just gets slapped into the Webpack somehow.

this would probably require major changes to the architecture of vencord anyway
ofc you'd also need to assume the user doesn't have node or git or pnpm
maybe a custom compiler could be used

although all of this is still risky considering how dumb discord users are, they will find a way to break it or do it wrong

**Comment by NeonGamerBot-QK** [comment]
Hey so i have one i made myself which runs all scripts at the `window.onload` event and runs gist scripts and snippets from a textbox. the ui is pretty bad though and dosent work on discord web. 
You can find it [here](https://github.com/NeonGamerBot-QK/Vencord/tree/main/src/plugins/run_js) *run at your OWN risk though*
